### PR TITLE
Add golangci-lint

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds a new golangci-lint target to the makefile and sets up
a common config along the lines of other openshift operators.

- Address misspell violations
- Address staticheck violations

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

### CODE REVIEW

**Comment by jlojosnegros** [comment]
/lgtm

cc @yanirq @MarSik

File: Makefile [hunk 1]
```diff
@@ -122,6 +124,10 @@ else
 	@exit 1
 endif
 
+golangci-lint:
```
Comment (yanirq) [inline Makefile:127]: I'm not sure we can directly use downloaded content here ,
@MarSik  WDYT?

Comment (jlojosnegros) [reply Makefile:127]: we are doing something similar with ginkgo for other tests ...

Comment (jlojosnegros) [reply Makefile:127]: @yanirq @MarSik 
kindly reminder

Comment (jmencak) [reply Makefile:127]: Downloading is fine as long as the downloading does not happen in the (ART) environment production (nightly/release) images are built.

**Review by jmencak — APPROVED** [review]
@danielmellado Thank you for the PR, I like it.  This should make the code cleaner.  It would be great if we didn't unnecessarily download the linter binary every time, e.g. we probably want a new download rule with the binary as a target in the makefile.  If you could make this change, it would be great.  Adding
/hold
because of that, otherwise
/lgtm
/approve

**Comment by ffromani** [comment]
I think to fix the failing lanes a rebase could be sufficient

**Comment by jlojosnegros** [comment]
@danielmellado I think this can be closed as #793 has been already merged.
Thanks for your help :)

**Pull request closed without merging**